dev/src/main/java/org/eclipse/codewind/intellij/module/CodewindModuleBuilder.java

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.codewind.intellij.module;
+
+import com.intellij.ide.util.projectWizard.*;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.module.*;
+import com.intellij.openapi.options.ConfigurationException;
+import com.intellij.openapi.progress.EmptyProgressIndicator;
+import com.intellij.openapi.roots.ModifiableRootModel;
+import com.intellij.openapi.roots.ui.configuration.ModulesProvider;
+import com.intellij.openapi.util.InvalidDataException;
+import org.eclipse.codewind.intellij.core.FileUtil;
+import org.eclipse.codewind.intellij.core.Logger;
+import org.eclipse.codewind.intellij.core.cli.InstallUtil;
+import org.eclipse.codewind.intellij.core.cli.ProjectUtil;
+import org.eclipse.codewind.intellij.core.connection.ProjectTemplateInfo;
+import org.jdom.JDOMException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeoutException;
+
+public class CodewindModuleBuilder extends JavaModuleBuilder {
+
+    private ProjectTemplateInfo template;
+
+    public CodewindModuleBuilder() {
+    }
+
+    @Override
+    public ModuleType getModuleType() {
+        return CodewindModuleType.CODEWIND_MODULE;
+    }
+
+    @Nullable
+    @Override
+    public ModuleWizardStep getCustomOptionsStep(WizardContext context, Disposable parentDisposable) {
+        try {
+            if (!InstallUtil.getInstallStatus().isInstalled()) {
+                return new InstallCodewindStep();
+            }
+            if (!InstallUtil.getInstallStatus().isStarted()) {
+                return new StartCodewindStep();
+            }
+            return createNewCodewindProjectStep();
+        } catch (Exception e) {
+            Logger.logWarning(e);
+        }
+        return null;
```

Comment:
I can't recall the state of the wizard buttons in your demo and whether there was a Previous button there.   What happened if you press previous and next in succession?